drud/ddev
# Add command to output final docker-compose config, closes #1192

## The Problem/Issue/Bug:
#1192: Debugging docker-compose configurations can be difficult when multiple docker-compose.*.yaml files exist, and many values are generated and inserted from environment variables. 

## How this PR Solves The Problem:
This PR adds a hidden debugging command, `ddev debug compose-config`, that will go through the same steps to create a full docker-compose configuration as when starting a project. After combining any docker-compose.*.yaml files and inserting environment variables, the command will output the final configuration state. This is especially helpful for determining the exact values that are passed to a container when debugging override files or trying to figure out exactly what values are available to the web/db/dba container.

## Manual Testing Instructions:
When in a project's approot, run `ddev debug compose-config` and ensure that a correct docker-compose configuration is printed.

- Update values like the project name, server type, etc. and ensure that these updates are propagated to the docker-compose config output
- Create a docker-compose.override.yaml and add additional values, override existing values, etc. and ensure that these updates are propagated to the docker-compose config output

Run `ddev debug compose-config <projectname>`, where <projectname> is the name of a running project, and ensure that a correct docker-compose configuration for the provided project has been printed

## Automated Testing Overview:
A functional test has been added that the function outputs expected values and that values from a docker-compose.override.yaml are included in the result.

## Related Issue Link(s):
#1192
